aptos-move/framework/aptos-stdlib/sources/string_utils.move

```diff
@@ -0,0 +1,146 @@
+/// A module for formatting move values as strings.
+module aptos_std::string_utils {
+    use std::string::String;
+
+    /// The number of values in the list does not match the number of "{}" in the format string.
+    const EARGS_MISMATCH: u64 = 1;
+    /// The format string is not valid.
+    const EINVALID_FORMAT: u64 = 2;
+
+    /// Format a move value as a human readable string,
+    /// eg. `to_string(&1u64) == "1"`, `to_string(&false) == "false"`, `to_string(&@0x1) == "@0x1"`.
+    /// For vectors and structs the format is similar to rust, eg.
+    /// `to_string(&cons(1,2)) == "Cons { car: 1, cdr: 2 }"` and `to_string(&vector[1, 2, 3]) == "[ 1, 2, 3 ]"`
+    /// For vectors of u8 the output is hex encoded, eg. `to_string(&vector[1u8, 2u8, 3u8]) == "0x010203"`
+    /// For std::string::String the output is the string itself including quotes, eg.
+    /// `to_string(&std::string::utf8(b"My string")) == "\"My string\""`
```

Comment:
It's so annoying to have the double quotes included. Why do we want to include that? @movekevin